File: 2015/05/php/01.php
<?php

declare(strict_types=1);

require_once("./common.php");

function main()
{
    $input = get_input_content();
    $niceCount = 0;

    foreach ($input as $s) {
        $result = check_model1($s);
        if ($result["result"] === "Nice") {
            $niceCount++;
        }
    }

    if ($niceCount !== 258) {
        throw new \Exception("Unexpected result: $niceCount");
    }
}

main();

File: 2015/05/rust/src/01.rs
use rust::{StringCheckResult, check_string_model1, read_input};

fn main() {
    let input = read_input();
    let mut nice_count = 0;

    for line in input.lines() {
        if check_string_model1(line) == StringCheckResult::Nice {
            nice_count += 1
        }
    }

    assert_eq!(nice_count, 258);
}
